chore: update README in the docs directory (#276)
oracle-samples/macaron: docs/README.md

# Macaron documentation
This directory contains the source for the documentation of Macaron hosted [here](https://oracle-samples.github.io/macaron/).

## Build the documentation

To build the documentation, please follow these steps:
1. Setup the dev environment for Macaron using the instructions [here](../README.md#getting-started).
2. Build the documentation by running this command from the root directory of Macaron:
```
make docs
```

This command will build and generate the documentation into `docs/_build/html`. To view it locally, run (with the dev environment activated):

```
python3 -m http.server -d docs/_build/html
```

## Developer guide

> ### The API reference.

Reviewer: ```suggestion
## Extend the API reference
```
And remove `## Developer guide`.

> The API reference is generated using the [sphinx-apidoc](https://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html) tool.

Reviewer: ```suggestion
If you add a new module, make sure that it is added to the API reference. The API reference is generated using the [sphinx-apidoc](https://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html) tool.
```

Author: Addressed in https://github.com/oracle-samples/macaron/pull/276/commits/4937b2d0f10affbec842dfa78e23213ec6a6a1d1

From within the root directory of Macaron, run (with the dev environment activated):
```
sphinx-apidoc --no-toc --module-first --force --maxdepth 1 --output-dir docs/source/pages/apidoc/ src/
```

> The API reference RST files will be generated into `docs/source/pages/apidoc/` where it will be picked up when we build the documentation using the instructions [above](#build-the-sphinx-documentation).

Reviewer: ```suggestion
This command will  generate the API reference RST files into `docs/source/pages/apidoc/`. Make sure to check in the changed source files to the repository.
```

Author: Addressed in https://github.com/oracle-samples/macaron/pull/276/commits/4937b2d0f10affbec842dfa78e23213ec6a6a1d1
